components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -843,12 +845,16 @@ public String createApplication(ApplicationModel applicationModel, String templa
     public void patchApplication(String applicationId, ApplicationPatchModel applicationPatchModel) {
 
         ServiceProvider appToUpdate = cloneApplication(applicationId);
+        String tenantDomain = ContextLoader.getTenantDomainFromContext();
         if (applicationPatchModel != null) {
+            if (applicationPatchModel.getName() != null && !applicationPatchModel.getName().equals(appToUpdate
```

Comment:
Better if we avoid having business logics in the API layer.
We can use service layer for that.